### DISCUSSION TITLE
Training on data that's labelled with a class and measurement

### DISCUSSION THREAD

**Discussion by grahamsz** [ultralytics/ultralytics#1245] [Q&A]
I'm trying to build something that can look at a container and assess how full it is. If i had a tagged training set that somehow econded "Glass" and "72% full" is there a way I could have yolo output a separate measure alongside the class and bbox?

**Comment by glenn-jocher** [comment]
@grahamsz hello! It's great to hear about your project. YOLOv8 specializes in object detection, which means it can identify and locate objects within an image, such as your "Glass" container. However, estimating the fill level, like "72% full," is a more nuanced task that typically falls under regression rather than classification or detection.

While YOLOv8 directly provides the class and bounding box, it doesn't natively output a continuous variable like a fill percentage. For your use case, you might consider a two-step approach:

1. Use YOLOv8 to detect the container and output its bounding box.
2. Pass the detected container region to a separate regression model trained to estimate the fill percentage.

This way, YOLOv8 handles the detection, and another model, possibly a custom neural network or even traditional image processing techniques, takes care of the fill estimation. You would need to train this second model on images of containers labeled with their fill percentages.

Remember to check out our documentation for more details on how to train and use YOLOv8 for object detection. Good luck with your project! 🚀
